### ISSUE TITLE
ZEVA - Sales Forecast Submission Table

### ISSUE THREAD

**Issue by shayjeff** [bcgov/zeva#2204]
**Describe the task**  
Create a new section in the model year report at the bottom of the Consumer ZEV Sales page in ZEVA. This section will include features for medium and large suppliers to upload and submit their ZEV sales forecasts using a standardized Excel template. It also allows for the manual entry of forecast data, displays detailed vehicle information, and calculates total sales forecasts.

**Purpose**  
The integration aims to streamline the process of submitting ZEV sales forecasts for compliance with ZEV Regulation (Act S. 18.1 and Reg S. 16.3). The collected data will aid in policy-making and planning to support the adoption of zero-emission vehicles. 

**Mock up** 
 https://www.figma.com/design/ijCyqDXXGVGDmmyfaXzPhk/ZEVA?node-id=18-202&t=OvSUrBDTY9QNmqjr-1
**Proto**
https://www.figma.com/proto/ijCyqDXXGVGDmmyfaXzPhk/ZEVA?page-id=50%3A193&node-id=18-202&viewport=153%2C-256%2C0.46&t=GOidMB0EBpgguc1q-1&scaling=scale-down-width&content-scaling=fixed&starting-point-node-id=18%3A202

- The mock-up demonstrates the layout, including a table for entering and reviewing forecast data, and sections for calculating total sales.  

  - **Data Entry and Review:** A table for viewing the uploaded data, including columns for model year, make, model, type, range, class, interior volume, and total sales forecast.
  - **Total Sales Calculation:** Automatic calculation of total sales for ICE vehicles, ZEVs, and overall vehicle sales for the required years displayed in that section. 

**Acceptance Criteria**  
- [ ] A table for reviewing ZEV sales forecast data, with fields for all necessary vehicle information.
- [ ] Automated calculation of total sales for ICE vehicles, ZEVs, and total vehicle sales for each year displayed in a summary section.
- [ ] Pagination or scrolling functionality to handle large data sets.
- [ ] Confirmation checkbox for suppliers to confirm the completeness and accuracy of their forecast submission.
- [ ] System functionality is tested and ready for demonstration to OEMs in a scheduled meeting in September.

**Additional context**  
- A meeting with the suppliers will be scheduled for the first week of September to present the new feature.
- The feature needs to be fully operational and released by the target date of October 1, 2024.

**Comment by ArawuSamuel1** [comment]
This should also be a separate card and should be done first 

- [ ] A downloadable Excel template provided for suppliers to fill out their forecasts.

**Comment by shayjeff** [comment]
The mock-up has been updated to remove the word "sales to supplied or vehicles supplier" Also, the the name of the report is now "Forecast Report."  I've noticed the card is still in progress so wasn't sure if another card was needed to capture these changes. If so, please let me know and I will create that card.

**Comment by katerinkus** [comment]
@tim738745 
- [x] A table for reviewing ZEV sales forecast data, with fields for all necessary vehicle information.
- [x]  Automated calculation of total sales for ICE vehicles, ZEVs, and total vehicle sales for each year displayed in a summary section.
- [x]  Pagination or scrolling functionality to handle large data sets.
Not very intuitive. I did not realize at first I had to press Enter after entering page number. Was this the intention or should it work differently?

- [x]  Confirmation checkbox for suppliers to confirm the completeness and accuracy of their forecast submission.
- [x]  System functionality is tested and ready for demonstration to OEMs in a scheduled meeting in September.
I am assuming this is referring to the demonstration that happened on the 5th of Sept.
